### ISSUE TITLE
[BUG] beantalkd queue maximum length can deal with?

### ISSUE THREAD

**Issue by shoother** [phalcon/cphalcon#11006]
The beantalkd official document:

>  - <bytes> is an integer indicating the size of the job body, not including the
   trailing "\r\n". This value must be less than max-job-size (default: 2**16).

beantalkd help:

```
Use: /usr/local/bin/beanstalkd [OPTIONS]

Options:
 -b DIR   wal directory
 -f MS    fsync at most once every MS milliseconds (use -f0 for "always fsync")
 -F       never fsync (default)
 -l ADDR  listen on address (default is 0.0.0.0)
 -p PORT  listen on port (default is 11300)
 -u USER  become user and group
 -z BYTES set the maximum job size in bytes (default is 65535)
 -s BYTES set the size of each wal file (default is 10485760)
            (will be rounded up to a multiple of 512 bytes)
 -c       compact the binlog (default)
 -n       do not compact the binlog
 -v       show version information
 -V       increase verbosity
 -h       show this help
```

So I run the beantalkd service by:

```
/usr/local/bin/beanstalkd -z 655350000000 -V
```

then I write a test script in lastest Phalcon version 2.0.7:

```php
try {
    // Connect to the queue
    $queue = new Phalcon\Queue\Beanstalk(
        array(
            'host' => '127.0.0.1',
            'port' => '11300'
        )
    );

    // Insert the job in the queue
    $jobId = $queue->put(
        array(
            'processVideo' => str_pad('0', 17000, 'o', STR_PAD_LEFT)
        )
    );

    var_dump($queue, $jobId);
}
catch (\Exception $e) {
    echo $e->getMessage()."\n";
}
echo "finished.\n";
exit;
```

worker script:

```php
// Connect to the queue
$queue = new Phalcon\Queue\Beanstalk(
    array(
        'host' => '127.0.0.1',
        'port' => '11300'
    )
);

while (($job = $queue->reserve())) {

    $message = $job->getBody();

    var_dump($message);

    $job->delete();
}
```

when insert job to queue, there is no problem.

```
object(Phalcon\Queue\Beanstalk)[79]
  protected '_connection' => resource(6, stream)
  protected '_parameters' => 
    array (size=2)
      'host' => string '127.0.0.1' (length=9)
      'port' => string '11300' (length=5)

string '4' (length=1)
finished.
```

but the worker don't receive the data and display an error message:

```
Notice: unserialize(): Error at offset 27 of 16366 bytes in /Applications/phalcon/scripts/beantalk.worker.php on line 11

Call Stack:
    0.0029     226848   1. {main}() /Applications/phalcon/scripts/beantalk.worker.php:0
    0.0032     227624   2. Phalcon\Queue\Beanstalk->reserve() /Applications/phalcon/scripts/beantalk.worker.php:11
    6.3532     253856   3. unserialize(???) /Applications/phalcon/scripts/beantalk.worker.php:11

bool(false)
```

who knows what's the problem?

Thanks.

**Comment by dercoder** [comment]
Issue is still there in phalcon. I tried https://github.com/pda/pheanstalk/ and its working fine.

**Comment by sergeyklay** [comment]
Can't reproduce.

```
$ php --ri phalcon

phalcon


Web framework delivered as a C-extension for PHP
phalcon => enabled
Author => Phalcon Team and contributors
Version => 2.0.10
Build Date => Jan  8 2016 03:02:54
Powered by Zephir => Version 0.9.1a-dev

Directive => Local Value => Master Value
phalcon.db.escape_identifiers => On => On
phalcon.db.force_casting => Off => Off
phalcon.orm.events => On => On
phalcon.orm.virtual_foreign_keys => On => On
phalcon.orm.column_renaming => On => On
phalcon.orm.not_null_validations => On => On
phalcon.orm.exception_on_failed_save => Off => Off
phalcon.orm.enable_literals => On => On
phalcon.orm.late_state_binding => Off => Off
phalcon.orm.enable_implicit_joins => On => On
phalcon.orm.cast_on_hydrate => Off => Off
phalcon.orm.ignore_unknown_columns => Off => Off
```

```sh
set -eufo pipefail

BEANSTALKD_VERSION="1.10"
src_url="https://github.com/kr/beanstalkd/archive/v${BEANSTALKD_VERSION}.tar.gz"

mkdir -p "$HOME/.local/bin"

curl -L "$src_url" | tar xz
pushd "beanstalkd-${BEANSTALKD_VERSION}"
    make
    mv beanstalkd "$HOME/.local/bin"
popd
rm -rf "beanstalkd-${BEANSTALKD_VERSION}"

cd "$HOME/.local/bin"
./beanstalkd -z 6553500000000000 -p 11301 -l 127.0.0.1 -V
# pid 13554
# bind 3 127.0.0.1:11301
# 
```

**Create jobs**
```php
use Phalcon\Queue\Beanstalk;

try {
    // Connect to the queue
    $queue = new Beanstalk(
        [
            'host' => '127.0.0.1',
            'port' => '11301'
        ]
    );

    // Insert the job in the queue
    $jobId2 = $queue->put(
        [
            'processVideo' => str_repeat('A', pow(2, 128))
        ]
    );

    // Insert the job in the queue
    $jobId1 = $queue->put(
        [
            'someJob' => str_pad('0', 128000, '0', STR_PAD_LEFT)
        ]
    );

    var_dump($queue, $jobId1, $jobId2);
}
catch (\Exception $e) {
    echo $e->getMessage(), PHP_EOL;
    exit($e->getCode());
}

echo 'Finished', PHP_EOL;
exit(0);
```

**Worker**
```php
use Phalcon\Queue\Beanstalk;

try {
    // Connect to the queue
    $queue = new Beanstalk(
        [
            'host' => '127.0.0.1',
            'port' => '11301'
        ]
    );

    while (($job = $queue->reserve())) {
        $message = $job->getBody();

        printf(
            "Serialized size: (%s byte)\nJson size: (%s byte)\n",
            mb_strlen(serialize($message), '8bit'),
            strlen(json_encode($message))
        );

        $job->delete();
    }
}
catch (\Exception $e) {
    echo $e->getMessage(), PHP_EOL;
    exit($e->getCode());
}

echo 'Finished', PHP_EOL;
exit(0);
```

**Creating jobs output**
```sh
$ php beanstalk.php 
class Phalcon\Queue\Beanstalk#1 (2) {
  protected $_connection =>
  resource(4) of type (stream)
  protected $_parameters =>
  array(2) {
    'host' =>
    string(9) "127.0.0.1"
    'port' =>
    string(5) "11301"
  }
}
string(1) "2"
string(1) "1"
Finished
```

**Worker output**
```sh
$ php worker.php 
Serialized size: (128032 byte)
Json size: (128014 byte)
```

**Comment by virgofx** [comment]
This was fixed with #11190 -- I bet @shoother is using earlier version?

**Comment by shoother** [comment]
@virgofx Thanks.
